Cargo.toml

```diff
@@ -10,23 +10,35 @@ homepage = "https://github.com/pingcap/raft-rs"
 documentation = "https://docs.rs/raft"
 description = "The rust language implementation of Raft algorithm."
 categories = ["algorithms", "database-implementations"]
+edition = "2018"
+build = "build.rs"
+
+[build-dependencies]
+protobuf-build = "0.4.2"
 
 [features]
 default = []
+gen = []
 failpoint = ["fail"]
 
 [dependencies]
 log = ">0.2"
-protobuf = "2.0.4"
+protobuf = "~2.1-2.2"
```

Comment:
Can we just use 2 here? The more precise the version spec, the more likely it is to have Cargo errors.